src/Native/Unix/System.Net.Security.Native/pal_gssapi.c

```diff
@@ -160,11 +160,11 @@ uint32_t NetSecurityNative_ImportTargetName(uint32_t* minorStatus,
 
     if (isNtlmTarget)
     {
-        nameType = GSS_C_NT_HOSTBASED_SERVICE;
+        nameType = (gss_OID)GSS_C_NT_HOSTBASED_SERVICE;
```

Comment:
What was the warning here? Looking at the definition of GSS_C_NT_HOSTBASED_SERVICE in gssapi.h, I can see:
```C
GSS_DLLIMP extern gss_OID GSS_C_NT_HOSTBASED_SERVICE;
```

Reply:
I think it may differ between macOS / Linux with `const` modifier. Would be nice to know though.

Reply:
Thanks for checking the header; I have double checked with gcc v5.5, this change is not required. I have reverted this line.

trivia:

* Without any changes in `pal_gssapi.c`, we get the following matrix work:
  * clang v3.9-onwards, gcc v6.0-onwards
* With `(gss_OID)GSS_KRB5_NT_PRINCIPAL_NAME` single cast:
  * clang v5.0-onwards, gcc v5.0-onwards
* With double cast `(gss_OID)(unsigned long)GSS_KRB5_NT_PRINCIPAL_NAME`:
  * clang v3.9-onwards, gcc v5.0-onwards

gcc v5 error without `gss_OID` cast:

```sh
[ 78%] Building C object System.Net.Security.Native/CMakeFiles/System.Net.Security.Native.dir/pal_gssapi.c.o
/corefx/src/Native/Unix/System.Net.Security.Native/pal_gssapi.c: In function 'NetSecurityNative_ImportTargetName':
/corefx/src/Native/Unix/System.Net.Security.Native/pal_gssapi.c:167:18: error: assignment discards 'const' qualifier from pointer target type [-Werror=discarded-qualifiers]
         nameType = GSS_KRB5_NT_PRINCIPAL_NAME;
                  ^
```

clang v4 error without `unsigned long` (additional) cast:

```sh
[ 77%] Building C object System.Net.Security.Native/CMakeFiles/System.Net.Security.Native.dir/pal_gssapi.c.o
/corefx/src/Native/Unix/System.Net.Security.Native/pal_gssapi.c:167:29: error: cast from 'const struct gss_OID_desc_struct *' to 'struct gss_OID_desc_struct *' drops const qualifier [-Werror,-Wcast-qual]
        nameType = (gss_OID)GSS_KRB5_NT_PRINCIPAL_NAME;
                            ^
```